src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ReferenceAssemblies+FileSystemSemaphore.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Testing
+{
+    public sealed partial class ReferenceAssemblies
+    {
+        private sealed class FileSystemSemaphore
```

Comment:
Can the filename have a `.` instead of a `+`?

Reply:
The current file name matches the type's metadata name. It also sorts better.